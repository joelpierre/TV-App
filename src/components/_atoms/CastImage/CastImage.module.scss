.CastImage {
  position: relative;
  width: 100%;
  border-radius: 50%;
  padding-bottom: 100%;
  overflow: hidden;
  background-color: $color-tint-gamma;

  &--no-image {
    &::after {
      @include center-xy;
      content: '';
      width: 12px;
      height: 12px;
      border-radius: 50%;
      background-color: $color-tint-alpha;
    }
  }

  &__image {
    @include center-xy;
    width: 100%;
    height: auto;
  }
}
