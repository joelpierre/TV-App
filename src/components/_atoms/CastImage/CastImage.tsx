import React from 'react';
import classNames from 'classnames';
import { ITvShowImage } from 'common/types/interfaces';

import styles from './CastImage.module.scss';

export interface ICastImageProps {
  className?: string;
  image: ITvShowImage;
  name: string;
}

const CastImage: React.FunctionComponent<ICastImageProps> = ({
  className,
  image,
  name
}) => {
  const imageSrc = image?.medium || image?.original;

  return (
    <div className={classNames(styles.CastImage, className, {
      [styles['CastImage--no-image']]: !image
    })}>
      {imageSrc && (
        <img
          src={imageSrc}
          className={styles.CastImage__image}
          alt={`${name} image`}
          width={58}
          loading="lazy"
        />
      )}
    </div>
  );
};

export default CastImage;
