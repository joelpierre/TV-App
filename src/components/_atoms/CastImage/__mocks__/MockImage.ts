import { ITvShowImage } from '../../../../types/interfaces';

export const MockImage: ITvShowImage = {
  medium: "http://static.tvmaze.com/uploads/images/medium_portrait/130/325669.jpg",
  original: "http://static.tvmaze.com/uploads/images/original_untouched/130/325669.jpg"
};
