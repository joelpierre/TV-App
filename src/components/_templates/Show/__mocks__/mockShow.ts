/* eslint-disable */
import { ITvShow } from '../../../../types/interfaces';
import { MockImage } from '../../../_atoms/CastImage/__mocks__/MockImage';

export const MockShow: ITvShow = {
  "id": 31534,
  "url": "http://www.tvmaze.com/shows/31534/the-russell-howard-hour",
  "name": "The Russell Howard Hour",
  "type": "Talk Show",
  "language": "English",
  "genres": [
    "Comedy"
  ],
  "status": "Running",
  "runtime": 60,
  "premiered": "2017-09-21",
  "officialSite": "https://www.sky.com/watch/title/series/89419c88-0079-426a-b29b-bc28bc403a58/the-russell-howard-hour",
  "schedule": {
    "time": "22:00",
    "days": [
      "Thursday"
    ]
  },
  "rating": {
    "average": 8.4
  },
  "weight": 83,
  "network": {
    "id": 63,
    "name": "Sky 1",
    "country": {
      "name": "United Kingdom",
      "code": "GB",
      "timezone": "Europe/London"
    }
  },
  "webChannel": null,
  "externals": {
    "tvrage": null,
    "thetvdb": 334902,
    "imdb": "tt7544016"
  },
  "image": MockImage,
  "summary": "<p><b>The Russell Howard Hour</b> is a weekly talk show in which Russell Howard will be chatting with celebrity guests alongside comedy stand-up. </p>",
  "updated": 1603404686,
  "_links": {
    "self": {
      "href": "http://api.tvmaze.com/shows/31534"
    },
    "previousepisode": {
      "href": "http://api.tvmaze.com/episodes/1955660"
    },
    "nextepisode": {
      "href": "http://api.tvmaze.com/episodes/1955661"
    }
  },
  "_embedded": {
    "cast": [
      {
        "person": {
          "id": 64866,
          "url": "http://www.tvmaze.com/people/64866/russell-howard",
          "name": "Russell Howard",
          "country": {
            "name": "United Kingdom",
            "code": "GB",
            "timezone": "Europe/London"
          },
          "birthday": "1980-03-23",
          "deathday": null,
          "gender": "Male",
          "image": {
            "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/175/437607.jpg",
            "original": "http://static.tvmaze.com/uploads/images/original_untouched/175/437607.jpg"
          },
          "_links": {
            "self": {
              "href": "http://api.tvmaze.com/people/64866"
            }
          }
        },
        "character": {
          "id": 456592,
          "url": "http://www.tvmaze.com/characters/456592/the-russell-howard-hour-host",
          "name": "Host",
          "image": null,
          "_links": {
            "self": {
              "href": "http://api.tvmaze.com/characters/456592"
            }
          }
        },
        "self": true,
        "voice": false
      }
    ]
  }
};
