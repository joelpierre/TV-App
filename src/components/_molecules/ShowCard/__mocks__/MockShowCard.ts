import { ITvSchedule } from '../../../../types/interfaces';
import { MockShow } from '../../../_templates/Show/__mocks__/mockShow';
import { MockImage } from '../../../_atoms/CastImage/__mocks__/MockImage';

export const MockShowCard: ITvSchedule = {
  url: '',
  type: '',
  summary: '',
  season: 2,
  runtime: 50,
  number: 2,
  name: 'Test Show Card',
  image: MockImage.original,
  id: 123,
  airtime: '09:00',
  airstamp: 'airStamp',
  airdate: '01-01-2020',
  show: MockShow
};
