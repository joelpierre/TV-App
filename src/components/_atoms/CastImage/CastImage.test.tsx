import React from 'react';
import { render, RenderResult } from 'test-utils';

import CastImage, { ICastImageProps } from './CastImage';
import { MockImage } from './__mocks__/MockImage';

const defaultProps: ICastImageProps = {
  name: 'Joel test',
  image: MockImage
};

const setup = (props: Partial<ICastImageProps> = {}): RenderResult => {
  const setupProps = { ...defaultProps, ...props };
  return render(<CastImage {...setupProps} />);
};

describe('<CastImage/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should contain image', () => {
    const img = wrapper.container.getElementsByTagName('img');
    expect(img).toHaveLength(1);
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  describe('Given no image is passed', () => {
    beforeEach(() => {
      wrapper = setup({ image: undefined });
    });

    it('Should render without crashing', () => {
      expect(wrapper.container).not.toBeEmptyDOMElement();
    });

    it('should not contain image', () => {
      const img = wrapper.container.getElementsByTagName('img');
      expect(img).toHaveLength(0);
    });
  });
});
