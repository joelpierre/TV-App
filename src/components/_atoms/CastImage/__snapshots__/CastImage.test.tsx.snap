// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CastImage/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="CastImage"
  >
    <img
      alt="Joel test image"
      class="CastImage__image"
      loading="lazy"
      src="//static.tvmaze.com/uploads/images/medium_portrait/130/325669.jpg"
      width="58"
    />
  </div>
</DocumentFragment>
`;
